content/articles/an-extensive-guide-on-working-with-images-in-django/index.md

```diff
@@ -0,0 +1,424 @@
+- [Adding a background image in a Django template](#adding-a-background-image-in-a-django-template)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+To follow through this tutorial practically, you need to:
+1. Have fundamental knowledge in both Python and Django
+2. Have a code editor.ie: VS code
+
+### Step 1 - Setting up the environment
+We shall start by setting up a development environment for our Django project. In your command line, create a new directory for our project and name it `Images`:
+```bash
+mkdir Images
+```
+
+Then enter the directory, create a virtual environment and activate it respectively using the commands below:
+```bash
+cd Images
+```
+```bash
+py -m venv .venv
+```
+```bash
+.venv\Scripts\activate.bat
+```
+Now we will need to install:
+ - Django - this python package helps build our website using the Django web development framework.
+ - Pillow - this is a python image library that we shall use to manipulate our images.
+
+We will install the above using:
+```bash
+pip install django
+```
+```bash
+pip install pillow
+```
+After the installation, we can now proceed to create the Django project named `myGallery`:
+```cd 
+django-admin startproject myGallery
+```
+Enter the project directory:
+```bash
+cd myGallery
+```
+and then create our app `demo`:
+```bash
+py manage.py startapp demo
+```
+Now we need to register our `demo` app in the list of installed apps. In the `settings.py` file under `installed apps`:
+
+```py
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'demo',                        # new
+]
+```
+Now we need to tell Django which URL to use when serving the media files and also give it a path to the root directory where our images will be stored. We shall do this by using `MEDIA_URL` and `MEDIA_ROOT` respectively. So while still in `settings.py` we shall add the following:
+
+```py
+MEDIA_URL = '/media/'
+MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
+
+```
+Also, make sure to import os at the top of `settings.py`
+```py
+import os
+```
+Now to enable a user to upload images during development, we need to add the following settings in our project-level `urls.py` for testing purposes. It should look like this:
+
+```py 
+from django.contrib import admin
+from django.urls import path
+from django.conf import settings
+from django.conf.urls.static import static 
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+
+] + static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
+
+```
+We are now done laying down the groundwork of uploading our images.
+
+### Step 2 - Creating and registering the models
+We can now proceed and define a model that shall be used for storing the images in our app. In `settings.py` we shall create a model with two fields, title of a photo and the photo itself:
```

Comment:
Add 1 - 2 lines of explanation about models. Some readers might be new to Django